src/NuGet.Core/NuGet.Protocol/HttpSource/HttpRetryHandler.cs

```diff
@@ -127,6 +163,18 @@ public class HttpRetryHandler : IHttpRetryHandler
                     {
                         response?.Dispose();
 
+                        ProtocolDiagnostics.RaiseEvent(new ProtocolDiagnosticEvent(
```

Comment:
This is after the entire chain of HTTP message handlers have run, right? Is this desired?

Reply:
I should have provided more context on the PR or at least the linked issue, but the ultimate aim is to get a feel for detecting slow package sources. It doesn't matter if a source is slow because a single HTTP resource was slow, or if there was many redirects before getting the resource. To the end-user, it still took the same amount of wall-clock time. That's all we need to measure.

Reply:
Okay, makes sense.

If a credential provider is waiting for user input does this get included in the time? Seems like that would still be part of the desired "wall time" so I guess that's what you want.

Reply:
If it's waiting for user input, I don't think it should count, but it's not something I've investigated. Thanks for bringing it up, I should look into it.

Reply:
We discussed this as well at some point in the last week or two...from looking at the code, the wait time for the auth might be included.

Reply:
Today's pushes fixed this issue.